// @aws-sdk/client-sns/dist-es/commands/CreateSMSSandboxPhoneNumberCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateSMSSandboxPhoneNumberInputFilterSensitiveLog, CreateSMSSandboxPhoneNumberResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryCreateSMSSandboxPhoneNumberCommand, serializeAws_queryCreateSMSSandboxPhoneNumberCommand, } from "../protocols/Aws_query";
export class CreateSMSSandboxPhoneNumberCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, CreateSMSSandboxPhoneNumberCommand.getEndpointParameterInstructions()));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SNSClient";
        const commandName = "CreateSMSSandboxPhoneNumberCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: CreateSMSSandboxPhoneNumberInputFilterSensitiveLog,
            outputFilterSensitiveLog: CreateSMSSandboxPhoneNumberResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryCreateSMSSandboxPhoneNumberCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryCreateSMSSandboxPhoneNumberCommand(output, context);
    }
}
