// @aws-sdk/client-sns/dist-es/commands/ListOriginationNumbersCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListOriginationNumbersRequestFilterSensitiveLog, ListOriginationNumbersResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryListOriginationNumbersCommand, serializeAws_queryListOriginationNumbersCommand, } from "../protocols/Aws_query";
export class ListOriginationNumbersCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, ListOriginationNumbersCommand.getEndpointParameterInstructions()));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SNSClient";
        const commandName = "ListOriginationNumbersCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: ListOriginationNumbersRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListOriginationNumbersResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryListOriginationNumbersCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryListOriginationNumbersCommand(output, context);
    }
}
